### PR TITLE
feat: add support for chart plugins

### PR DESCRIPTION
**Pull request by veinfors**
### Render with plugins
```js
embed(app).render({
  element,
  type: 'my-chart',
  plugins: [plugin]
});
```

### usePlugins hook
```js
import { usePlugins } from '@nebula.js/stardust';
// ...
const plugins = usePlugins();
plugins.forEach((plugin) => {
  // Invoke plugin
  plugin.fn();
});
```

### Plugin format
```js
const plugin = {
  info: {
    name: "example-plugin",
    type: "meta-type",
  },
  fn: () => {
    // Plugin implementation goes here
  }
};
```

### CODE REVIEW

File: apis/nucleus/src/plugins/plugins.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/**
+ * @interface Plugin
+ * @property {object} info Object that can hold various meta info about the plugin
+ * @property {string} info.name The name of the plugin
+ * @property {function} fn The implementation of the plugin. Input and return value of this function can be whatever is necessery for This can be called from the chart implementation with necessary input.
```
Comment (Caele) [inline apis/nucleus/src/plugins/plugins.js:5]: - Add description (just a line of text at the top)
- Fix fn description to be readable.

**Comment by Caele** [comment]
Looks good I think. Can we add a .md file to the docs here as well with a light example? That way we have something to bring over to the official docs as we release this.

**Comment by veinfors** [comment]
> Looks good I think. Can we add a .md file to the docs here as well with a light example? That way we have something to bring over to the official docs as we release this.

Sure, good idea, I will come up with some example and att that.

File: apis/nucleus/src/plugins/plugins.js (hunk 1 above)
Comment (veinfors) [reply apis/nucleus/src/plugins/plugins.js]: Done and done!

**Comment by veinfors** [comment]
> > Looks good I think. Can we add a .md file to the docs here as well with a light example? That way we have something to bring over to the official docs as we release this.
> 
> Sure, good idea, I will come up with some example and add that.

@Caele I've added a `plugins.md` file... Take a look if it's OK or if something is missing...

**Review by Caele — APPROVED** [review]

**Pull request merged**